ganga/GangaDirac/Lib/Files/DiracFile.py

```diff
@@ -403,6 +403,49 @@ def remove(self):
         self.guid = ''
         return True
 
+    @require_credential
+    def massRemove(self, files):
+        """
+        A method to remove many files at once for maximum speed by making fewer calls to DIRAC.
+        Do this in blocks of 200 to prevent timeouts.
+        files can be a list of DiracFile or a list of lfn strings.
+        """
+        lfnsToRemove = []
+        for _f in files:
+            if isinstance(_f, DiracFile) and not _f.lfn=="":
+                lfnsToRemove.append(_f.lfn)
+            elif isinstance(_f, str):
+                lfnsToRemove.append(_f)
+        i = 0
+        _successes = []
+        _failures = []
+        while i < len(lfnsToRemove):
+            dirac_cmd = "removeFile(%s)" % lfnsToRemove[i:i+200]
+            try:
+                result = execute(dirac_cmd, cred_req=self.credential_requirements)
+            except GangaDiracError as err:
+                msg = 'Problem removing files: %s' % str(err)
+                logger.warning(msg)
+                return False
+            #Check that the number of removed files is the same as the number requested
+            if result.get('Successful', False):
+                _successes.extend(result['Successful'].keys())
+            if result.get('Failed', False):
+                _failures.extend(result['Failed'].keys())
+            i = i+200
+
+        def clearFileInfo(f):
```

Comment:
You are changing information of objects that might sit inside a completed job. Will the job get correctly marked as dirty and the change persisted?

Reply:
I'm not sure how one would mark the job. The change of the DiracFile object should be persisted though.

Reply:
If it works, I am not too worried.